### DISCUSSION TITLE
Awesomewm

### DISCUSSION THREAD

**Discussion by LeBaronDeCharlus** [catppuccin/catppuccin#2181] [Port Requests]
**Are you available to help in the creation of this port? 😄 (optional)**
Yes

**Additional context 📷 (optional)**
Porting awesomewm catppuccin theme :smiley: 

Preview repository available here : https://github.com/Kaderovski/catppuccin-awesomemwm

![image](https://user-images.githubusercontent.com/91135191/176874085-a6fb2b6f-0699-46f5-b735-9ade40a20ba3.png)

Edit : No more issue while trying to join Discord

**Comment by nekowinston** [comment]
Lovely, don't use awesomewm myself, but just saw your Discord error; very weird.
Might be an issue on Discords end? People are still actively joining the server at the moment.

Maybe this works: https://discord.gg/nP3fbRy5

**Comment by LeBaronDeCharlus** [comment]
Any updates on it ? 🤗

**Comment by nekowinston** [comment]
Thanks for your patience on this. I've recently become a member of the org with merge access; I will check it out this weekend for myself. If all goes well, I'll see that it gets transferred ASAP.

**Comment by nekowinston** [comment]
One thing that's still missing would be the implementation of other flavours, such as Latte.
Since WezTerm is also lua based, you might want to have a look at [our port](https://github.com/catppuccin/wezterm/blob/65078e846c8751e9b4837a575deb0745f0c0512f/catppuccin.lua#L5-L31) for it, how colours are defined there.

I've also chatted with @Isabelincorp about this port, I believe they've been working on implementing other flavours.

**Comment by nonetrix** [comment]
I am unable to get it working without installing Lain might want to put this in the docs

No not this Lain:
![863edb6e016fb21fd8413e320c74f336d75123f4_hq](https://user-images.githubusercontent.com/45698918/179925891-f4f64c71-0a90-4d0d-9388-d7758032c6fb.jpg)

This Lain:
https://github.com/lcpz/lain

**Comment by ffflick-gh** [comment]
I've implemented catppuccin for my own awesomewm setup before along with all four flavours; perhaps it might be useful: https://github.com/quintik/turbo-awesome/tree/master/themes/catppuccin.

It can then be used like so:
```lua
local theme = require "themes.catppuccin.[flavour]"
local beautiful = require "beautiful"
beautiful.init(theme)
```

@Kaderovski perhaps remove `require("lain")`? If the user doesn't have it installed it would raise an error, and it doesn't seem like `lain` is being used anywhere in `theme.lua`

**Comment by ghost** [comment]
Ah, I have been messing with awesomewm again, but I would suggest just having a copy and paste section for ctp (all 4 flavours) to throw into your theme file because of how preference based the theme file is.
We don't want to end up with a [rofi](https://github.com/catppuccin/rofi) situation
